File: src/main/java/epicsquid/superiorshields/enchantment/RagingEnchantment.java
package epicsquid.superiorshields.enchantment;

import epicsquid.superiorshields.capability.shield.IShieldCapability;
import epicsquid.superiorshields.config.Config;
import net.minecraft.world.item.enchantment.EnchantmentCategory;

public class RagingEnchantment extends DamageBoostEnchantment {

	public RagingEnchantment(Rarity rarityIn, EnchantmentCategory typeIn) {
		super(rarityIn, typeIn);
	}

	@Override
	public int getMaxLevel() {
		return 1;
	}

	@Override
	public float boostDamage(float damage) {
		return damage + Config.SHIELD.RAGING_DAMAGE_ADDED.get().floatValue();
	}

	@Override
	public boolean shouldBoostDamage(IShieldCapability shield) {
		return shield.getCurrentHp() <= 0;
	}

	@Override
	public int getMinCost(int pEnchantmentLevel) {
		return 1;
	}

	@Override
	public int getMaxCost(int pEnchantmentLevel) {
		return this.getMinCost(pEnchantmentLevel) + 40;
	}
}

File: src/main/kotlin/dev/epicsquid/superiorshields/enchantment/RagingEnchantment.kt
package dev.epicsquid.superiorshields.enchantment

import dev.epicsquid.superiorshields.config.Config
import dev.epicsquid.superiorshields.effects.EffectHandler
import dev.epicsquid.superiorshields.registry.CapabilityRegistry.shield
import net.minecraft.world.entity.LivingEntity
import net.minecraft.world.item.enchantment.EnchantmentCategory

class RagingEnchantment(
	rarity: Rarity,
	category: EnchantmentCategory,
	effectHandler: EffectHandler
) : DamageBoostEnchantment(rarity, category, effectHandler) {

	override fun getMaxLevel(): Int = 1

	override fun getMinCost(level: Int): Int = 1

	override fun getMaxCost(level: Int): Int = super.getMinCost(level) + 40

	override fun boostDamage(damage: Float): Float =
		damage + Config.SHIELDS_CONFIG.ragingDamageAdded.get().toFloat()

	override fun shouldBoostDamage(entity: LivingEntity): Boolean =
		entity.shield.hp <= 0


}